addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -70,12 +70,14 @@ def _check_move_configuration(self, move):
         error_message += self._l10n_in_validate_partner(move.company_id.partner_id, is_company=True)
         if not re.match("^.{1,16}$", move.name):
             error_message.append(_("Invoice number should not be more than 16 characters"))
-        for line in move.invoice_line_ids.filtered(lambda line: line.display_type not in ('line_note', 'line_section', 'rounding')):
+        for index, line in enumerate(move.invoice_line_ids.filtered(lambda line: line.display_type not in ('line_note', 'line_section', 'rounding')), start=1):
```

Comment:
```suggestion
        for line in move.invoice_line_ids.filtered(lambda line: line.display_type not in ('line_note', 'line_section', 'rounding')):
``` Not needed

Reply:
An index is necessary because, in cases where there are multiple lines with the same product and a negative discount, an index will be useful for easy navigation of those lines.

[Refer this](https://www.awesomescreenshot.com/image/42877546?key=c11de83a00560b6d5a6307d0cf0c0b91)

Reply:
@kasp-odoo There are plenty of places in Odoo where we refer to lines and we almost never invent a line number on the fly.  And it is also a bit weird to have an invoice with twice the same product.  Also, the product_id is not mandatory.

Reply:
@jco-odoo for Indian EDI product is mendetory because of hsn code.
And in some business same product with multiple is common because price is changed by lots and seasons.
So it's like one with old price and another one is with new price.
So it's useful to navigate problematic lines.
So what you suggest in this?

Reply:
Oh, it's added here, I think it's already there.
@kasp-odoo remove this from here we can discuss and add in master if needed.

Reply:
@jva-odoo okay, got it.